jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -293,8 +314,11 @@ public void fail(Throwable t)
             try
             {
                 getContentSource().fail(t);
-                if (temporary)
-                    delete();
+                try (AutoLock ignored = lock.lock())
+                {
+                    if (temporary)
+                        delete();
```

Comment:
As there are only a handful filesystem operations, I'm leaning towards doing them out of locking.
Can we have a `private void delete(Path)` called outside locks that is called after `this.path = null`?

Reply:
Done.